File: dev/permissions.py
from jam.auth import Permissions

# TODO make user specific
OSF = 'user-osf-*'

ADMIN = 'tracked-experimenter|admins-*'
PARTICIPANT = 'tracked-experimenter|accounts-*'

config = [
    (ADMIN, Permissions.CRUD),
    (OSF, Permissions.CRUD)
]

admin = [
    (ADMIN, Permissions.READ),
    (OSF, Permissions.READ),
    (ADMIN, Permissions.CREATE),
    (OSF, Permissions.CREATE)
]

experiment = [
    (ADMIN, Permissions.READ),
    (OSF, Permissions.READ),
    (ADMIN, Permissions.CREATE),
    (OSF, Permissions.CREATE),
    (PARTICIPANT, Permissions.READ)
]

session = [
    (PARTICIPANT, Permissions.CREATE),
    (ADMIN, Permissions.READ),
    (OSF, Permissions.READ)
]

account = [
    (ADMIN, Permissions.CREATE),
    (OSF, Permissions.CREATE),
    (ADMIN, Permissions.READ),
    (OSF, Permissions.READ)
]

profile = [
    (PARTICIPANT, Permissions.CREATE),
    (ADMIN, Permissions.READ),
    (OSF, Permissions.READ)
]

File: scripts/permissions.js
// TODO make user specific
var PARTICIPANT = 'jam-experimenter:accounts-*';
var GLOBAL = '*';

var config = [
];

var experiment = [
    [GLOBAL, 'READ'],
];

var session = [
    [PARTICIPANT, 'CREATE'],
    [PARTICIPANT, 'UPDATE'],
];

var account = [
    [GLOBAL, 'CREATE'],
];

var profile = [
    [PARTICIPANT, 'CREATE'],
];

var thumbnail = [
    [PARTICIPANT, 'READ'],
];

module.exports = {
    account: account,
    //config: config,
    experiment: experiment,
    profile: profile,
    thumbnail: thumbnail,
    session: session
};
